packet/conn.go

```diff
@@ -97,10 +96,25 @@ func (c *Conn) ReadPacketReuseMem(dst []byte) ([]byte, error) {
 
 	if err := c.ReadPacketTo(buf); err != nil {
 		return nil, errors.Trace(err)
+	}
+
+	readBytes := buf.Bytes()
+	readSize := len(readBytes) + len(dst)
+	result := make([]byte, 0, readSize)
```

Comment:
No. It doesn't look like reusing of memory :)

What about leaving the logic in BytesBufferPut, but removing it here?

Reply:
I make some change again.
always cache big buffer cause too much memory usage.

Reply:
Maybe it will be enough for us leave the old version
```go
result = append(dst, buf.Bytes()...)
```
but change
```go
defer utils.BytesBufferPut(buf)
```
to
```go
defer func(){
	if readSize <= utils.TooBigBlockSize { // Or move this logic into utils.BytesBufferPut itself?
		utils.BytesBufferPut(buf)
	}
}()
```

Reply:
bytes in buffer would be copied once
```
result := append(dst, buf.Bytes()...)
```
but if not put it to buffer pool again, do not need copy it any more.
```
		if readSize > utils.TooBigBlockSize {
			// if read block is big, use read block as result and do not cache buf any more
			result = readBytes
			buf = nil

		} else {
			result = append(dst, readBytes...)
		}
```

Reply:
Ok, I understood. And agree.

Could you fix golangci warnings?

Reply:
OK, done.